src/bci_build/package/appcontainers.py

```diff
@@ -808,3 +809,82 @@ def _get_nginx_kwargs(os_version: OsVersion):
     )
     for tomcat_major, os_version in product(_TOMCAT_VERSIONS, ALL_BASE_OS_VERSIONS)
 ]
+
+SPACK_CONTAINERS = [
+    ApplicationStackContainer(
+        name="spack",
+        package_name="spack-image",
+        pretty_name="Spack package manager for supercomputers - build container",
+        os_version=os_version,
+        is_latest=os_version in CAN_BE_LATEST_OS_VERSION,
+        other_tags=["next"]
+        if not (
+            (os_version in CAN_BE_LATEST_OS_VERSION)
+            or (os_version in RELEASED_OS_VERSIONS)
+        )
+        else None,
+        version="%%spack_version%%",
+        version_in_uid=False,
+        replacements_via_service=[
+            Replacement(
+                regex_in_build_description="%%spack_version%%",
+                package_name="spack",
+                parse_version="patch",
```

Comment:
With this, the container tag will be `:0.21` and not the full version including the patch release
```suggestion
                parse_version="minor",
```